### PR TITLE
Operators for arrays

### LINKED ISSUE #722: Operators for Arrays
**Is your feature request related to a problem? Please describe.**
Currently, we have many nice array math functions like `add`, `sub`, ... However, they are ugly to use and nest.

**Describe the solution you'd like**
Overloaded and templated operators that use these functions.

**Describe alternatives you've considered**
Keep living with the pain.

**Additional context**
- Would make code more readable.
- When introducing this we should also replace any usage of these functions with the new operators.

### PR DESCRIPTION
**Pull request by dmartin13**
# Description

For the most common array operations (`add`, `sub`, `mul`, `div`, `addScalar`, `subScalar`, `mulScalar`) there are now operators defined within the namespace ```autopas::utils::ArrayMath::literals;```. Almost all occurrences of these ArrayMath functions in AutoPas have been replaced by the corresponding operators.

## Resolved Issues

- fixes #722 

# How Has This Been Tested?

All existing tests were executed. And new test:

- ArrayMathTest/testAddOpInt
- ArrayMathTest/testAddOpDouble
- ArrayMathTest/testMulOpDouble
- ArrayMathTest/testAddScalarOp
- ArrayMathTest/testSubScalarOp
- ArrayMathTest/testMulScalarOp

### CODE REVIEW

File: examples/sph-mpi/sph-main-mpi.cpp [hunk 1]
```diff
@@ -409,10 +415,12 @@ void densityPressureHydroForce(AutoPasContainer &sphSystem, MPI_Comm &comm, cons
 }
 
 void printConservativeVariables(AutoPasContainer &sphSystem, MPI_Comm &comm) {
+  using namespace autopas::utils::ArrayMath::literals;
+
   std::array<double, 3> momSum = {0., 0., 0.};  // total momentum
   double energySum = 0.;                        // total energy
   for (auto it = sphSystem.begin(autopas::IteratorBehavior::owned); it.isValid(); ++it) {
-    momSum = autopas::utils::ArrayMath::add(momSum, autopas::utils::ArrayMath::mulScalar(it->getV(), it->getMass()));
+    momSum = momSum + (it->getV() * it->getMass());
```
Comment (FG-TUM) [inline examples/sph-mpi/sph-main-mpi.cpp:423]: you could also implement [assignment operators](https://en.cppreference.com/w/cpp/language/operator_assignment)

File: src/autopas/containers/linkedCells/traversals/LCC04Traversal.h [hunk 2]
```diff
@@ -46,7 +46,7 @@ class LCC04Traversal : public C08BasedTraversal<ParticleCell, PairwiseFunctor, d
                                                                                  interactionLength, cellLength),
         _cellOffsets32Pack(computeOffsets32Pack()),
         _cellHandler(pairwiseFunctor, this->_cellsPerDimension, interactionLength, cellLength, this->_overlap),
-        _end(utils::ArrayMath::subScalar(utils::ArrayUtils::static_cast_array<long>(this->_cellsPerDimension), 1l)) {}
+        _end(utils::ArrayMath::operator-(utils::ArrayUtils::static_cast_array<long>(this->_cellsPerDimension), 1l)) {}
```
Comment (FG-TUM) [inline src/autopas/containers/linkedCells/traversals/LCC04Traversal.h:49]: I'm not really convinced that this change improves readability.

File: src/autopas/containers/linkedCells/traversals/LCC04SoACellHandler.h [hunk 3]
```diff
@@ -389,9 +392,9 @@ inline void LCC04SoACellHandler<ParticleCell, PairwiseFunctor, dataLayout, useNe
         // back right
         if (y != _overlap[1] and x != _overlap[0] and z != 0) {
           // check whether cell is within cutoff radius
-          auto distVec = utils::ArrayMath::mul(
-              {std::max(0.0, _overlap[0] - x - 1.0), std::max(0.0, _overlap[1] - y - 1.0), std::max(0.0, z - 1.0)},
-              _cellLength);
+          auto distVec = std::array<double, 3>{std::max(0.0, _overlap[0] - x - 1.0),
```
Comment (FG-TUM) [inline src/autopas/containers/linkedCells/traversals/LCC04SoACellHandler.h:395]:
```suggestion
          const auto distVec = std::array<double, 3>{std::max(0.0, _overlap[0] - x - 1.0),
```

File: tests/testAutopas/tests/utils/ArrayMathTest.cpp [hunk 4]
```diff
@@ -34,6 +34,32 @@ TEST(ArrayMathTest, testAddDouble) {
...
 }
 
+TEST(ArrayMathTest, testAddOpInt) {
+  using namespace autopas::utils::ArrayMath::literals;
+  std::array<int, 3> a({1, 2, 3}), b({4, 5, 6});
+  std::array<int, 3> correctResult({5, 7, 9});
+  std::array<int, 3> result = a + b;
+  for (int d = 0; d < 3; ++d) {
+    ASSERT_EQ(result[d], correctResult[d]);
+  }
```
Comment (FG-TUM) [inline tests/testAutopas/tests/utils/ArrayMathTest.cpp:44]: I'm not a fan of hardcoded results. It's way more safe, understandable, and resiliant to changes to the test if you calculate the correct results in the test. E.g. like this:
```suggestion
  std::array<int, 3> a({1, 2, 3}), b({4, 5, 6});
  std::array<int, 3> result = a + b;
  for (int d = 0; d < 3; ++d) {
    ASSERT_EQ(result[d], a[d] + b[d]);
  }
```

File: src/autopas/containers/linkedCells/traversals/LCC04Traversal.h (hunk 2 above)
Comment (dmartin13) [reply src/autopas/containers/linkedCells/traversals/LCC04Traversal.h]: done in [fcbb30b](https://github.com/AutoPas/AutoPas/pull/735/commits/fcbb30b047a00acac67d3090e735f6141ffc5da8).

**Pull request merged**